internal/controller/eventing/integrationtests/controller/integration_test.go

```diff
@@ -0,0 +1,297 @@
+package controller_test
+
+import (
+	"os"
+	"testing"
+
+	eventingv1alpha1 "github.com/kyma-project/eventing-manager/api/v1alpha1"
+	eventingcontroller "github.com/kyma-project/eventing-manager/internal/controller/eventing"
+	"github.com/kyma-project/eventing-manager/test/matchers"
+	"github.com/kyma-project/eventing-manager/test/utils"
+	testutils "github.com/kyma-project/eventing-manager/test/utils/integration"
+	natsv1alpha1 "github.com/kyma-project/nats-manager/api/v1alpha1"
+	"github.com/onsi/gomega"
+	gomegatypes "github.com/onsi/gomega/types"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/apps/v1"
+
+	ecdeployment "github.com/kyma-project/kyma/components/eventing-controller/pkg/deployment"
+	natstestutils "github.com/kyma-project/nats-manager/testutils"
+)
+
+const projectRootDir = "../../../../../"
+
+var testEnvironment *testutils.TestEnvironment //nolint:gochecknoglobals // used in tests
+
+// TestMain pre-hook and post-hook to run before and after all tests.
+func TestMain(m *testing.M) {
+	// Note: The setup will provision a single K8s env and
+	// all the tests need to create and use a separate namespace
+
+	// setup env test
+	var err error
+	testEnvironment, err = testutils.NewTestEnvironment(projectRootDir, false, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// run tests
+	code := m.Run()
+
+	// tear down test env
+	if err = testEnvironment.TearDown(); err != nil {
+		panic(err)
+	}
+
+	os.Exit(code)
+}
+
+func Test_CreateEventingCR(t *testing.T) {
+	// t.Parallel()
+
+	testCases := []struct {
+		name                 string
+		givenEventing        *eventingv1alpha1.Eventing
+		givenNATS            *natsv1alpha1.NATS
+		givenDeploymentReady bool
+		givenNATSReady       bool
+		wantMatches          gomegatypes.GomegaMatcher
+	}{
+		{
+			name: "Eventing CR should have processing due to NATS is not available",
+			givenEventing: utils.NewEventingCR(
+				utils.WithEventingCRNamespace(ecdeployment.PublisherNamespace),
+				utils.WithEventingCRDefaults(),
+				utils.WithEventingStreamData("Memory", "1M", "1M", 1, 1),
+				utils.WithEventingPublisherData(1, 1, "199m", "99Mi", "399m", "199Mi"),
+			),
+			givenNATS: natstestutils.NewNATSCR(
+				natstestutils.WithNATSCRNamespace(ecdeployment.PublisherNamespace),
+				natstestutils.WithNATSCRDefaults(),
+			),
+			givenNATSReady: false,
+			wantMatches: gomega.And(
+				matchers.HaveStatusError(),
+				matchers.HaveNATSAvailableConditionNotAvailable(),
+			),
```

Comment:
Test says `Eventing CR should have processing`, but matchers check for `matchers.HaveStatusError()`.